src/Utilities/Extensions/ImmutableHashSetBuilderExtensions.cs

```diff
@@ -10,31 +11,70 @@ namespace Analyzer.Utilities.Extensions
     internal static class ImmutableHashSetBuilderExtensions
     {
         // Just to make hardcoding SinkInfos more convenient.
-        public static void AddSink(
+        public static void AddSinkInfo(
             this ImmutableHashSet<SinkInfo>.Builder builder,
             string fullTypeName,
             SinkKind sinkKind,
             bool isInterface,
             bool isAnyStringParameterInConstructorASink,
             IEnumerable<string> sinkProperties,
             IEnumerable<(string Method, string[] Parameters)> sinkMethodParameters)
+        {
+            builder.AddSinkInfo(
+                fullTypeName,
+                new[] { sinkKind },
+                isInterface, 
+                isAnyStringParameterInConstructorASink,
+                sinkProperties,
+                sinkMethodParameters);
+        }
+
+        // Just to make hardcoding SinkInfos more convenient.
+        public static void AddSinkInfo(
+            this ImmutableHashSet<SinkInfo>.Builder builder,
+            string fullTypeName,
+            IEnumerable<SinkKind> sinkKinds,
+            bool isInterface,
+            bool isAnyStringParameterInConstructorASink,
+            IEnumerable<string> sinkProperties,
+            IEnumerable<(string Method, string[] Parameters)> sinkMethodParameters)
         {
             SinkInfo sinkInfo = new SinkInfo(
                 fullTypeName,
-                sinkKind,
+                sinkKinds.ToImmutableHashSet(),
                 isInterface,
                 isAnyStringParameterInConstructorASink,
                 sinkProperties:
                     sinkProperties != null
-                        ? sinkProperties.ToImmutableHashSet()
+                        ? sinkProperties.ToImmutableHashSet(StringComparer.Ordinal)
                         : ImmutableHashSet<string>.Empty,
                 sinkMethodParameters:
                     sinkMethodParameters != null
                         ? sinkMethodParameters
                              .Select(o => new KeyValuePair<string, ImmutableHashSet<string>>(o.Method, o.Parameters.ToImmutableHashSet()))
-                             .ToImmutableDictionary()
+                             .ToImmutableDictionary(StringComparer.Ordinal)
                         : ImmutableDictionary<string, ImmutableHashSet<string>>.Empty);
             builder.Add(sinkInfo);
         }
+
+        // Just to make hardcoding SourceInfos more convenient.
+        public static void AddSourceInfo(
+            this ImmutableHashSet<SourceInfo>.Builder builder,
+            string fullTypeName,
+            bool isInterface,
+            string[] taintedProperties,
+            string[] taintedMethods)
+        {
+            SourceInfo metadata = new SourceInfo(
+                fullTypeName,
+                isInterface: isInterface,
+                taintedProperties: taintedProperties != null
```

Comment:
same suggestion as above - null coalesce properties and supply a default for when the null shortcut is taken